File: src/main/java/my/study/repository/design/pattern/state/java/PowerOffState.java
package my.study.repository.design.pattern.state.java;

/**
 * @author djh on  2018/11/12 22:08
 * @E-Mail 1544579459@qq.com
 */
public class PowerOffState implements State {
    @Override
    public void nextChannel() {
        System.out.println("关机状态,无法操作");
    }

    @Override
    public void turnUpVolume() {
        System.out.println("关机状态,无法操作");
    }

    @Override
    public void turnDownVolume() {
        System.out.println("关机状态,无法操作");
    }
}

File: src/main/java/my/study/repository/design/pattern/state/kotlin/PowerOffState.kt
package my.study.repository.design.pattern.state.kotlin

/**
 * @author djh on  2018/11/12 22:24
 * @E-Mail 1544579459@qq.com
 */
class PowerOffState : State {
    override fun nextChannel() {
        println("关机状态,无法操作")
    }
    
    override fun turnUpVolume() {
        println("关机状态,无法操作")
    }
    
    override fun turnDownVolume() {
        println("关机状态,无法操作")
    }
}